db/models.py

```diff
@@ -50,3 +57,83 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        to=settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        related_name="user_orders"
+    )
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        date_str = self.created_at.strftime("%Y-%m-%d")
+        time_str = self.created_at.strftime("%H:%M:%S")
+        return f"{date_str} {time_str}"
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(
+        to=MovieSession,
+        on_delete=models.CASCADE,
+        related_name="movie_session_tickets"
+    )
+    order = models.ForeignKey(
+        to=Order,
+        on_delete=models.CASCADE,
+        related_name="order_tickets"
+    )
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    def clean(self) -> None:
+        rows = self.movie_session.cinema_hall.rows
+        seats_in_row = self.movie_session.cinema_hall.seats_in_row
+
+        if not (1 <= self.row <= rows):
+            raise ValidationError(
+                {
+                    "row": [
+                        f"row number must be in available range: "
+                        f"(1, rows): (1, {rows})"
+                    ]
+                }
+            )
+
+        if not (1 <= self.seat <= seats_in_row):
+            raise ValidationError(
+                {
+                    "seat": [
+                        f"seat number must be in available range: "
+                        f"(1, seats_in_row): (1, {seats_in_row})"
+                    ]
+                }
+            )
+
+    def save(self, *args, **kwargs) -> Optional[None]:
+        self.full_clean()
+        return super().save(*args, **kwargs)
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(
+                fields=["row", "seat", "movie_session"],
+                name="unique_ticket_row_seat_movie_session"
+            )
+        ]
+
+    def __str__(self) -> str:
+        date_str = self.movie_session.show_time.strftime("%Y-%m-%d")
+        time_str = self.movie_session.show_time.strftime("%H:%M:%S")
+        return (
+            f"{self.movie_session.movie.title} {date_str} {time_str} "
+            f"(row: {self.row}, seat: {self.seat})"
+        )
+
+
+class User(AbstractUser):
+    pass
```

Comment:
Ensure that the custom `User` model is correctly configured in the Django settings if it is intended to replace the default user model. This involves setting `AUTH_USER_MODEL = 'yourapp.User'` in the settings file.